opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDateType.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.opensearch.data.type;
+
+import static org.opensearch.common.time.DateFormatter.splitCombinedPatterns;
+import static org.opensearch.common.time.DateFormatter.strip8Prefix;
+import static org.opensearch.sql.data.type.ExprCoreType.DATE;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.EqualsAndHashCode;
+import org.opensearch.common.time.DateFormatter;
+import org.opensearch.common.time.FormatNames;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+
+/**
+ * Date type with support for predefined and custom formats read from the index mapping.
+ */
+@EqualsAndHashCode(callSuper = true)
+public class OpenSearchDateType extends OpenSearchDataType {
+
+  private static final OpenSearchDateType instance = new OpenSearchDateType();
+
+  private static final String FORMAT_DELIMITER = "\\|\\|";
```

Comment:
Maybe
```suggestion
  private static final String FORMAT_DELIMITER_REGEX = "\\|\\|";
```

Reply:
UPD it is never used - remove?

Reply:
Done in https://github.com/Bit-Quill/opensearch-project-sql/pull/268/commits/58a8f193ffe1407665734d55a60e53ff9bc97aa8